mleap-core/src/main/scala/ml/combust/mleap/core/feature/CountVectorizerModel.scala

```diff
@@ -27,15 +36,34 @@ case class CountVectorizerModel(vocabulary: Array[String],
         }
         tokenCount += 1
     }
-
+    val valuesArray = mutable.ArrayBuilder.make[Double]
+    val indicesArray = mutable.ArrayBuilder.make[Seq[Int]]
     val effectiveMinTF = if (minTf >= 1.0) minTf else tokenCount * minTf
-    val effectiveCounts = if(binary) {
-      termCounts.filter(_._2 >= effectiveMinTF).map(p => (p._1, 1.0)).toSeq
-    } else {
-      termCounts.filter(_._2 >= effectiveMinTF).toSeq
+    val iterator = termCounts.iterator
+    if (binary){
+      while (iterator.hasNext){
+        val (termIndex, count) = iterator.next()
+        if (count>= effectiveMinTF){
+          valuesArray += 1.0
+          indicesArray += seqCache(termIndex)
+        }
+      }
+    }else{
+      while (iterator.hasNext){
+        val (termIndex, count) = iterator.next()
+        if (count>= effectiveMinTF){
+          valuesArray += count
+          indicesArray += seqCache(termIndex)
+        }
+      }
```

Comment:
while loops are faster than filter + map. This also eliminates the need to create a Seq of tuples and then unzip or iterate over it to get indices and values